cpp/ql/src/semmle/code/cpp/commons/unix/Constants.qll

```diff
@@ -15,44 +20,77 @@ int parseOctal(string input) {
     )
 }
 
+/** Gets the number corresponding to the set-user-ID on execute bit in Unix. */
 int s_isuid() { result = parseOctal("04000") }
 
+/** Gets the number corresponding to the set-group-ID on execute bit in Unix. */
 int s_isgid() { result = parseOctal("02000") }
 
+/** Gets the number corresponding to the sticky bit in Unix. */
 int s_isvtx() { result = parseOctal("01000") }
 
+/** Gets the number corresponding to the read permission bit for owner of the file in Unix. */
 int s_irusr() { result = parseOctal("0400") }
 
+/** Gets the number corresponding to the write permission bit for owner of the file in Unix. */
 int s_iwusr() { result = parseOctal("0200") }
 
+/** Gets the number corresponding to the execute permission bit for owner of the file in Unix. */
 int s_ixusr() { result = parseOctal("0100") }
 
+/** Gets the number corresponding to the permissions `S_IRUSR | S_IWUSR | S_IXUSR` in Unix. */
 int s_irwxu() { result = s_irusr().bitOr(s_iwusr()).bitOr(s_ixusr()) }
 
+/**
+ * Gets the number corresponding to the read permission bit for the group
+ * owner of the file in Unix.
+ */
 int s_irgrp() { result = s_irusr().bitShiftRight(3) }
 
+/**
+ * Gets the number corresponding to the write permission bit for the group
+ * owner of the file in Unix.
+ */
 int s_iwgrp() { result = s_iwusr().bitShiftRight(3) }
 
+/**
+ * Gets the number corresponding to the execute permission bit for the group
+ * owner of the file in Unix.
+ */
 int s_ixgrp() { result = s_ixusr().bitShiftRight(3) }
 
+/** Gets the number corresponding to the permissions `S_IRGRP | S_IWGRP | S_IXGRP` in Unix. */
 int s_irwxg() { result = s_irwxu().bitShiftRight(3) }
 
+/** Gets the number corresponding to the read permission bit for other users in Unix. */
 int s_iroth() { result = s_irgrp().bitShiftRight(3) }
 
+/** Gets the number corresponding to the write permission bit for other users in Unix. */
 int s_iwoth() { result = s_iwgrp().bitShiftRight(3) }
 
+/** Gets the number corresponding to the execute-or-search permission bit for other users in Unix. */
 int s_ixoth() { result = s_ixgrp().bitShiftRight(3) }
 
+/** Gets the number corresponding to the permissions `S_IROTH | S_IWOTH | S_IXOTH` in Unix. */
 int s_irwxo() { result = s_irwxg().bitShiftRight(3) }
 
+/**
+ * Gets the number that can be used in a bitwise and with the file status flag
+ * to produce a number representing the file access mode.
```

Comment:
```suggestion
 * to produce a number representing the file access mode on Linux.
```

This number and all the numbers below are Linux-specific, right? Or do they also work on Mac/BSD? The POSIX standard doesn't specify these numbers but actually [remarks that the numbers used on Linux were a mistake](https://pubs.opengroup.org/onlinepubs/9699919799/functions/open.html#tag_16_357_08).

Reply:
The values of `O_RDONLY`, `O_WRONLY`, `O_RDWR` and `O_ACCMODE` on my machine and on BSD (https://github.com/freebsd/freebsd/blob/master/sys/sys/fcntl.h#L75) and linux (https://github.com/torvalds/linux/blob/master/include/uapi/asm-generic/fcntl.h) all match the numbers in the .qll file. But indeed, the values of `O_CREAT` and `O_EXCL` are Linux specific.

Reply:
I'm fine with being pragmatic here and making it only the final 2.